File: frontend/src/models/Project.model.ts
import { Dayjs } from 'dayjs';

import { type IPattern } from '@/models/Pattern.model';

export interface IProject {
    readonly pattern: IPattern;
    readonly purchasedAt: Dayjs;
    readonly pausePositionX: number | null;
    readonly pausePositionY: number | null;
}

File: backend/ClickStitch/Models/ProjectModel.cs
namespace ClickStitch.Models;

public sealed class ProjectModel
{
    public required PatternModel Pattern { get; init; }
    public required Guid Reference { get; init; }
    public required DateTime PurchasedAt { get; init; }
    public required int? PausePositionX { get; init; }
    public required int? PausePositionY { get; init; }
}